packages/api/src/command/medical/cq-directory/get-cq-directory-entry.ts

```diff
@@ -18,3 +18,20 @@ export const getCQDirectoryEntryOrFail = async ({
   if (!organization) throw new NotFoundError(`Could not find CQ organization`, undefined, { oid });
   return organization;
 };
+
+export const getCQDirectoryEntriesIdsAndLastUpdated = async (
+  oids: string[]
+): Promise<{ id: string; oid: string; lastUpdated: string }[]> => {
+  const entries = await CQDirectoryEntryModel.findAll({
+    where: {
+      oid: oids,
+    },
```

Comment:
add: 
`attributes: ['id', 'oid', 'lastUpdated'],`
to the search to improve efficiency by only retrieving necessary fields. In the spirit of a suggestion Raf has been making to explicitly decompose patient: { id, cxId } in function headers